### PR TITLE
Support OVN-IC on DPU

### PR DESCRIPTION
**Pull request by wizhaoredhat**

### CODE REVIEW

File: controllers/config_controller.go [hunk 1]
```diff
@@ -245,11 +270,32 @@ func (r *DpuClusterConfigReconciler) syncOvnkubeDaemonSet(ctx context.Context, c
 	}
 
 	data := render.MakeRenderData()
+
+	// TODO: KUBE_RBAC_PROXY_IMAGE should be specified when running the operator...
+	// in CNO it's defined in the YAML for CNO itself:
+	// - name: KUBE_RBAC_PROXY_IMAGE
+	//   value: "quay.io/openshift/origin-kube-rbac-proxy:latest"
+	// https://github.com/openshift/cluster-network-operator/blob/master/manifests/0000_70_cluster-network-operator_03_deployment.yaml#L69-L70
+	data.Data["KubeRBACProxyImage"] = "registry.redhat.io/openshift4/ose-kube-rbac-proxy" // os.Getenv("KUBE_RBAC_PROXY_IMAGE")
```
Comment (ricky-rav) [inline controllers/config_controller.go]: we can remove this comment of mine :)

File: controllers/config_controller.go [hunk 2]
```diff
@@ -245,11 +270,32 @@ func (r *DpuClusterConfigReconciler) syncOvnkubeDaemonSet(ctx context.Context, c
...
+	data.Data["LISTEN_DUAL_STACK"] = listenDualStack(masterIPs[0])
+
+	if len(masterIPs) == 1 {
+		data.Data["NorthdThreads"] = 1
+	} else {
+		// OVN 22.06 and later support multiple northd threads.
+		// Less resource constrained clusters can use multiple threads
+		// in northd to improve network operation latency at the cost
+		// of a bit of CPU.
+		data.Data["NorthdThreads"] = 4
```
Comment (ricky-rav) [inline controllers/config_controller.go]: we went back to having 1 thread only for northd when deployed with ovn IC:
https://github.com/openshift/cluster-network-operator/commit/703ccb516dc5eef5a5d108f2458e31c54f258428

File: controllers/config_controller.go (hunk 1 above)
Comment (wizhaoredhat) [reply controllers/config_controller.go]: Done

File: controllers/config_controller.go (hunk 2 above)
Comment (wizhaoredhat) [reply controllers/config_controller.go]: Thanks for sharing =)

**Pull request closed without merging**